gui/layers/_markers.py

```diff
@@ -0,0 +1,218 @@
+        """float, None: width of the symbol edge in px
+        """
+
+        return self._edge_width
+
+    @edge_width.setter
+    def edge_width(self, edge_width):
+        self._edge_width = edge_width
+        self.refresh()
+
+    @property
+    def edge_width_rel(self):
+        """float, None: width of the marker edge as a fraction
+            of the marker size.
+
+            vispy docs say: "exactly one edge_width and
+            edge_width_rel must be supplied", but I don't know
+            what that means... -KY
+        """
+
+        return self._edge_width_rel
+
+    @edge_width_rel.setter
+    def edge_width_rel(self, edge_width_rel):
+        self._edge_width_rel = edge_width_rel
+        self.refresh()
+
+    @property
+    def edge_color(self):
+        """Color, ColorArray: the marker edge color
+        """
+
+        return self._edge_color
+
+    @edge_color.setter
+    def edge_color(self, edge_color):
+        self._edge_color = edge_color
+        self.refresh()
+
+    @property
+    def face_color(self):
+        """Color, ColorArray: color of the body of the marker
+        """
+
+        return self._face_color
+
+    @face_color.setter
+    def face_color(self, face_color):
+        self._face_color = face_color
+        self.refresh()
+
+    @property
+    def scaling(self):
+        """bool: if True, marker rescales when zooming
+        """
+
+        return self._scaling
+
+    @scaling.setter
+    def scaling(self, scaling):
+        self._scaling = scaling
+        self.refresh()
+
+    def _get_shape(self):
+
+        return np.max(self.marker_coords, axis=0) + 1
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+
+            self.viewer._child_layer_changed = True
+            self.viewer._update()
+
+            self._set_view_slice(self.viewer.indices)
+
+        if self._need_visual_update:
+            self._need_visual_update = False
+            self._node.update()
+
+    def _refresh(self):
+        """Fully refresh the underlying visual.
+        """
+        self._need_display_update = True
+        self._update()
+
+    def _set_view_slice(self, indices):
+        """Sets the view given the indices to slice with.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
+ 
+        # Get a list of the coords for the markers in this slice
+        coords = self.marker_coords
+        matches = np.equal(coords[:, 2:],
+                   np.repeat([indices[2:]], len(coords), axis=0))
+        in_slice_markers = coords[np.all(matches, axis=1), :2]
+
+        # Display markers if there are any in this slice
+        if len(in_slice_markers) > 0:
+            # Get the marker sizes
+            if isinstance(self.size, np.ndarray):
+                sizes = self.size[np.all(matches, axis=1)]
```

Comment:
If you’re going to reuse `np.all(...)` you should assign it to a variable. Perhaps `matches` since it goes unused.